### ISSUE TITLE
utility to setup TLS to make the first developer experience nice

### ISSUE THREAD

**Issue by patriknw** [akka/akka-grpc#89]
Either support plain text mode as the Netty client is already doing, or have a utility for setting up dummy test certificates on both sides.

Related https://github.com/akka/akka-http/issues/1849

**Comment by ignasi35** [comment]
> or have a utility for setting up dummy test certificates on both sides.

Akka could ship fake certs on the testkit, or there could be a tool to crete a CA and a server cert on the fly.

For Lagom we spiked the second alternative and built a [`FakeKeyStoreGenerator.scala`](https://github.com/ignasi35/lagom/commit/12bbd8a99947cfe3858235336635284c507d8cef#diff-7b40ab76a5e229e3fe091531bef2421f9) based on Play's [`FakeKeyStore`](https://github.com/playframework/playframework/blob/5f576efda61d9a1bd4cd27215e1fe221ddbaa3a9/framework/src/play-server/src/main/scala/play/core/server/ssl/FakeKeyStore.scala).

**Comment by raboof** [comment]
I think @richdougherty has been working on that, seeing if it is reasonable to use ssl-config here or we can make it convenient some other way.

**Comment by ktoso** [comment]
Right, I think our approach here should be "whichever style play already does it for dev" :)
If ssl config helps then good, but if not then don't force it in I think. It may be nice to use just to declare from where to load certs, but not expose it in any APIs unless needed perhaps for example.

**Comment by patriknw** [comment]
Remember to remove the certs from Quickstarts also, e.g. https://github.com/akka/akka-grpc-quickstart-scala.g8/pull/1#discussion_r202044228

**Comment by ktoso** [comment]
The comment to do what play does still stands, but they what play does does not work very well now: https://github.com/playframework/playframework/issues/8562

**Comment by ktoso** [comment]
For plain Akka to generate the certs we'd need to "steal" code that play has for these things I suppose:

from here

Rich Dougherty: 
https://github.com/playframework/playframework/blob/master/framework/src/play-server/src/main/scala/play/core/server/ssl/FakeKeyStore.scala#L53-L82

Rich Dougherty:
https://github.com/playframework/playframework/blob/master/framework/src/play-server/src/main/scala/play/core/server/ssl/DefaultSSLEngineProvider.scala#L62

**Comment by ignasi35** [comment]
raised https://github.com/akka/akka-grpc-quickstart-java.g8/issues/39

**Comment by ignasi35** [comment]
(sorry, pressed the wrong buttons)

**Comment by johanandren** [comment]
Not entirely sure this is still valid/important, but, related developments #1910

**Comment by johanandren** [comment]
We do support non-tls and samples and quickstarts has self-signed certs.

**Comment by johanandren** [comment]
We should update the quickstarts to use the new cert loading APIs when that is out.
